skimage/registration/_optical_flow.py

```diff
@@ -217,3 +218,152 @@ def optical_flow_tvl1(reference_image, moving_image,
                      tol=tol, prefilter=prefilter)
 
     return coarse_to_fine(reference_image, moving_image, solver, dtype=dtype)
+
+
+def _ilk(reference_image, moving_image, flow0, radius, num_warp, gaussian,
+         prefilter):
+    """Iterative Lucas-Kanade (iLK) solver for optical flow estimation.
+
+    Parameters
+    ----------
+    reference_image : ndarray, shape (M, N[, P[, ...]])
+        The first gray scale image of the sequence.
+    moving_image : ndarray, shape (M, N[, P[, ...]])
+        The second gray scale image of the sequence.
+    flow0 : ndarray, shape (reference_image.ndim, M, N[, P[, ...]])
+        Initialization for the vector field.
+    radius : int
+        Radius of the window considered around each pixel.
+    num_warp : int
+        Number of times moving_image is warped.
+    gaussian : bool
+        if True, a gaussian kernel is used for the local
+        integration. Otherwise, a uniform kernel is used.
+    prefilter : bool
+        Whether to prefilter the estimated optical flow before each
+        image warp. This helps to remove potential outliers.
+
+    Returns
+    -------
+    flow : ndarray, shape ((reference_image.ndim, M, N[, P[, ...]])
+        The estimated optical flow components for each axis.
+
+    """
+    dtype = reference_image.dtype
+    ndim = reference_image.ndim
+    size = 2 * radius + 1
+
+    if gaussian:
+        sigma = (0, ) + ndim * (size / 4, )
+        filter_func = partial(ndi.gaussian_filter, sigma=sigma, mode='mirror')
+    else:
+        filter_func = partial(ndi.uniform_filter, size=ndim * (size, ),
+                              mode='mirror')
+
+    flow = flow0
+    # For each pixel location (i, j), the optical flow X = flow[:, i, j]
+    # is the solution of the ndim x ndim linear system
+    # A[i, j] * X = b[i, j]
+    A = np.zeros(reference_image.shape + (ndim, ndim), dtype=dtype)
+    b = np.zeros(reference_image.shape + (ndim, ), dtype=dtype)
+
+    grid = np.meshgrid(*[np.arange(n, dtype=dtype)
+                         for n in reference_image.shape],
+                       indexing='ij', sparse=True)
+
+    for _ in range(num_warp):
+        if prefilter:
+            flow = ndi.filters.median_filter(flow, (1, ) + ndim * (3, ))
+
+        moving_image_warp = warp(moving_image, get_warp_points(grid, flow),
+                                 mode='nearest')
+        grad = np.array(np.gradient(moving_image_warp))
```

Comment:
```suggestion
        grad = np.stack(np.gradient(moving_image_warp), axis=0)
```
Nitpick: array stacking style consistent with #4826